### <font color="blue"> Sales Balance Data - Confidence Interval and Margin of Error Analysis </font> ###


In [2]:
import pandas as pd

# Reading Sales balance data Excel file into a Data Frame
balance_data = pd.read_excel('Balance.xlsx')

# Balance info
print("Shape of Balance data: ")
print("(No.of Rows, No.of Columns): ", balance_data.shape)
print("\n")

print("First three rows of data set, for quick verification: \n")
balance_data.head(3)

Shape of Balance data: 
(No.of Rows, No.of Columns):  (70, 1)


First three rows of data set, for quick verification: 



,New Balance
0,9430
1,7535
2,4078


### <font color="blue"> Descriptive Statistics </font> ###

In [15]:
import numpy as np
from scipy import stats

new_balance = balance_data['New Balance']

# Calculate basic descriptive statistics
stats_summary = new_balance.describe()

# Calculate required descriptive statistics parameters
mean = new_balance.mean()
std_err = new_balance.sem()  # Standard Error
std_dev = new_balance.std()
variance = new_balance.var()
kurtosis = new_balance.kurtosis()
skewness = new_balance.skew()
range_val = new_balance.max() - new_balance.min()
sum_val = new_balance.sum()
count = new_balance.count()

# Calculate Confidence Level (Margin of Error) at 95%
confidence = 0.95
degrees_freedom = count - 1
# Calculate Margin of Error
margin_of_error = stats.t.ppf((1 + confidence) / 2, degrees_freedom) * std_err

# Result Parameters
results = pd.Series({
    "Mean": mean,
    "Standard Error": std_err,
    "Median": new_balance.median(),
    "Mode": new_balance.mode()[0] if not new_balance.mode().empty else np.nan,
    "Standard Deviation": std_dev,
    "Sample Variance": variance,
    "Kurtosis": kurtosis,
    "Skewness": skewness,
    "Range": range_val,
    "Minimum": new_balance.min(),
    "Maximum": new_balance.max(),
    "Sum": sum_val,
    "Count": count,
    "Confidence Level (95.0%)": margin_of_error
})

# Format the display of results
display_results = results.copy()

int_rows = ['Count', 'Mode', 'Sum', 'Range', 'Minimum', 'Maximum']

formatted_output = display_results.apply(
    lambda x: f"{int(x)}" if display_results.index[display_results.tolist().index(x)] in int_rows else f"{x:.4f}"
)

print("Descriptive Statistics\n")
print(formatted_output)

Descriptive Statistics

Mean                            9312.0000
Standard Error                   478.9281
Median                          9466.0000
Mode                                13627
Standard Deviation              4006.9998
Sample Variance             16056047.7971
Kurtosis                          -0.2960
Skewness                           0.1879
Range                               18648
Minimum                               615
Maximum                             19263
Sum                                651840
Count                                  70
Confidence Level (95.0%)         955.4354
dtype: object


### <font color="blue"> Confidence Intervals </font> ###

In [19]:
LCI = mean - margin_of_error
UCI = mean + margin_of_error
print("Confidence Intervals\n")
print(f"Lower Bound (LCI): {LCI:.2f}")
print(f"Upper Bound (UCI): {UCI:.2f}")

Confidence Intervals

Lower Bound (LCI): 8356.56
Upper Bound (UCI): 10267.44


### <font color="blue"> Inference </font> ###

Based on the descriptive statistics of the 'New Balance' sample data, we can be 95% confident that the true population parameter, Mean will fall between 8356.56 and 10267.44, as the Margin of Error is 955.44. 